### ISSUE TITLE
How to evaluate the results

### ISSUE THREAD

**Issue by Mullerio** [loftusa/owls#3]
Hello,

thanks for the interested paper. Currently, I am trying to understand how you were able to generate the final.csv for the results, I have been running the animals.py script for some other models, which works well, but when trying to evaluate these results I am having problems. 

For example, when looking at the correlation between logit values and the corresponding subliminal prompting values, they seem to be uncorrelated. This holds for multiple models i have tried. 

<img width="4770" height="5895" alt="Image" src="https://github.com/user-attachments/assets/cba03675-8665-44e0-93a6-e7ad39a50109" />

For more results you can see https://github.com/LouisYRYJ/influence-animal-numbers/blob/main/jonas/results.

I would appreciate some guidance on how the evaluation should be done or if you could share the code you did it with (or point out how it can be done using the existing code).

Thank you very much in advance.
Best regards 
Jonas

**Comment by AmirZur** [comment]
Hi,

Thank you for expressing interest in our research!

These correlation plots look about right to me. The correlation between logits and subliminal prompting success certainly isn't high, but it's significant for a number of animals. In Table 1 of [our paper](https://openreview.net/pdf?id=auKgpBRzIW) we report the number of animals where the Pearson correlation is significant (< 0.05). 

The `final.csv` file is for Figure 2, where we plot the probability of generating each animal with & without subliminal prompting. We report the best subliminal prompting result from the top 10 numbers ranked by each method. Here's a code chunk for generating the `logit_scores` column for `final.csv`:
```
import pandas as pd
from utils.animals_utils import get_numbers, get_animals

N = 10

model_name ="Qwen/Qwen2.5-7B-Instruct" # "google/gemma-2-9b-it" # "meta-llama/Llama-3.1-8B-Instruct"
model_nickname = model_name.split('/')[-1]
subliminal_prompting_df = pd.read_csv(f"results/{model_nickname}/subliminal_prompting.csv", index_col=0)
subliminal_prompting_df['number'] = get_numbers()

logit_df = pd.read_csv(f'results/{model_nickname}/logit.csv', index_col=0)
logit_df['number'] = get_numbers()

logit_scores = []
logit_numbers = []
for animal, _ in get_animals(model_name):
    top_n = logit_df.sort_values(by=animal, ascending=False).iloc[:N].number
    top_n_df = subliminal_prompting_df[subliminal_prompting_df['number'].isin(top_n)]
    score = top_n_df[animal].max()
    logit_number = top_n_df.iloc[top_n_df[animal].argmax()]['number']
    logit_scores.append(score)
    logit_numbers.append(logit_number)

[f"{n}: {s:.2f}" for n, s in zip(logit_numbers, logit_scores)]
``` 

The `logit_scores` column stores the log probability for each animal when subliminal prompting with the number chosen by the logit method. 

Hope this helps!
